### PR TITLE
Ivy/di/inherited services

### PR DESCRIPTION
**Pull request by alxhub**

### CODE REVIEW

File: packages/core/src/render3/di.ts [hunk 1]
```diff
@@ -637,8 +637,8 @@ export class NodeInjector implements Injector {
 /**
  * @codeGenApi
  */
-export function ɵɵgetFactoryOf<T>(type: Type<any>): ((type: Type<T>| null) => T)|null {
-  const typeAny = type as any;
+export function ɵɵgetFactoryOf<T>(type: Type<any>): FactoryFn<T>|null {
+  const typeAny = `type ` as any;
```
Comment (gkalpak) [inline packages/core/src/render3/di.ts]: Is this intentional? It looks weird :grin:

Comment (alxhub) [reply packages/core/src/render3/di.ts]: Definitely not intentional, haha. Thanks for catching!

**Review by mhevery — APPROVED** [review]

**Comment by alxhub** [comment]
[Presubmit](https://test.corp.google.com/ui#id=OCL:251511177:BASE:252117320:1559942216770:9be37a72)

[Ivy Presubmit](http://test/OCL:251511177:BASE:252136514:1559948338752:95d3cc16)

File: packages/core/src/di/interface/defs.ts [hunk 2]
```diff
@@ -181,7 +190,15 @@ export function ɵɵdefineInjector(options: {factory: () => any, providers?: any
...
 export function getInjectableDef<T>(type: any): ɵɵInjectableDef<T>|null {
-  return type && type.hasOwnProperty(NG_INJECTABLE_DEF) ? type[NG_INJECTABLE_DEF] : null;
+  const def = type[NG_INJECTABLE_DEF] as ɵɵInjectableDef<T>;
+  // The definition read above may come from a base class. `hasOwnProperty` is not sufficient to
+  // distinguish this case, as in older browsers (e.g. IE10) static property inheritance is
+  // implemented by copying the properties.
+  //
+  // Instead, the ngInjectableDef's token is compared to the type, and if they don't match then the
+  // property was not defined directly on the type itself, and was likely inherited. The definition
+  // is only returned if the type matches the def.token.
```
Comment (kara) [inline packages/core/src/di/interface/defs.ts]: Thanks for the comment 👍

**Review by kara — APPROVED** [review]
LGTM

**Pull request closed without merging**